src/sentry/utils/batching_kafka_consumer.py

```diff
@@ -177,18 +216,33 @@ def __init__(
         if queued_max_messages_kbytes is None:
             queued_max_messages_kbytes = DEFAULT_QUEUED_MAX_MESSAGE_KBYTES
 
-        self.consumer = self.create_consumer(
-            topics,
-            cluster_name,
-            group_id,
-            auto_offset_reset,
-            queued_max_messages_kbytes,
-            queued_min_messages,
-        )
-
         self.producer = producer
         self.commit_log_topic = commit_log_topic
         self.dead_letter_topic = dead_letter_topic
+        if not consumer:
+            self.consumer = self.create_consumer(
+                topics,
+                cluster_name,
+                group_id,
+                auto_offset_reset,
+                queued_max_messages_kbytes,
+                queued_min_messages,
+            )
+        else:
+            assert isinstance(consumer, KafkaConsumerFacade)
+            self.consumer = consumer
+
+        def on_partitions_assigned(consumer, partitions):
+            logger.info("New partitions assigned: %r", partitions)
```

Comment:
Isn't the post process forwarder going to need to provide its own `on_partition_assigned` and `on_partition_revoked` methods ?

Reply:
I am not sure if its needed. I will handle this case during the integration PR if needed. Some context. 

The batching kafka consumer calls
self.consumer.subscribe at https://github.com/getsentry/sentry/pull/28677/files#diff-0baa7d2f6a7029fbcf066aacd1590491a129a3e20a6e7444005aca32b0ad0778R243-R245

Which in turn calls https://github.com/getsentry/sentry/blob/master/src/sentry/eventstream/kafka/consumer.py#L266-L325 This has its own assignment and revocation methods. Those methods call the additional callbacks provided as arguments above.

The flush method in the revocation callback at https://github.com/getsentry/sentry/pull/28677/files#diff-0baa7d2f6a7029fbcf066aacd1590491a129a3e20a6e7444005aca32b0ad0778R241 calls self._commit at https://github.com/getsentry/sentry/blob/master/src/sentry/utils/batching_kafka_consumer.py#L387 which calls the consumer commit method here https://github.com/getsentry/sentry/blob/master/src/sentry/utils/batching_kafka_consumer.py#L401 which takes care of committing the offsets.

I will do some additional testing though during the integration though to validate things